controller/coq_lsp.ml

```diff
@@ -203,31 +229,29 @@ let do_change params =
     assert false
   | change :: _ ->
     let contents = string_field "text" change in
-    let doc = Hashtbl.find doc_table uri in
-    let doc =
-      (* [bump_version] will clean stale info about the document, in particular
-         partial results of a previous checking *)
-      if version > doc.version then (
-        LIO.log_info "bump file" (uri ^ " / version: " ^ string_of_int version);
-        let tb = Unix.gettimeofday () in
-        let doc = Fleche.Doc.bump_version ~version ~contents doc in
-        let diff = Unix.gettimeofday () -. tb in
-        LIO.log_info "bump file took" (Format.asprintf "%f" diff);
-        doc)
-      else doc
-    in
-    let () = Hashtbl.replace doc_table uri doc in
-    Check.schedule ~uri
+    let doc = DocHandle.find_doc ~uri in
+    if version > doc.version then (
+      LIO.log_info "bump file" (uri ^ " / version: " ^ string_of_int version);
+      let tb = Unix.gettimeofday () in
+      let doc = Fleche.Doc.bump_version ~version ~contents doc in
+      let diff = Unix.gettimeofday () -. tb in
+      LIO.log_info "bump file took" (Format.asprintf "%f" diff);
+      let () = DocHandle.update_doc_version ~doc in
+      Check.schedule ~uri)
+    else
+      (* That's a weird case, get got changes without a version bump?
```

Comment:
Shall we log this in case we encounter it somehow?

Reply:
Indeed.